### PR TITLE
Allow passing custom clickhouse settings

### PR DESCRIPTION
**Pull request by macobo**
## Description

This PR fixes https://github.com/PostHog/charts-clickhouse/issues/249 by making clickhouse settings configurable.

Companion documentation PR: https://github.com/PostHog/posthog.com/pull/2833

## Type of change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How has this been tested?

See tests. I also tested manually. One edge case I ran into was when passing invalid settings, clickhouse doesn't automatically update. If then killed, the pod won't start up again until the issue is fixed.

## Checklist
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: charts/posthog/templates/clickhouse_instance.yaml [hunk 1]
```diff
@@ -13,6 +13,9 @@ spec:
 
     profiles:
       default/allow_experimental_window_functions: "1"
```
Comment (guidoiaquinti) [inline charts/posthog/templates/clickhouse_instance.yaml:15]: As discussed on Slack, let's put this as a common default in `values.yaml` and then merge that map/dict with `.Values.clickhouse.profiles`

File: ci/kubetest/test_clickhouse_custom_settings_profile.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+@pytest.fixture
+def setup(kube):
+    cleanup_k8s()
+    install_chart(VALUES_YAML)
+    wait_for_pods_to_be_ready(kube)
+
+
+def test_custom_clickhouse_settings_profile(setup, kube):
+    command = f"clickhouse-client -q \"SELECT name, value FROM system.settings WHERE name IN ('max_execution_time', 'max_memory_usage') ORDER BY name FORMAT JSON\""
+    result = kubectl_exec("chi-posthog-posthog-0-0-0", command)
```
Comment (guidoiaquinti) [inline ci/kubetest/test_clickhouse_custom_settings_profile.py:54]: Nit: not sure about it but `kubetest` might expose a method on the container object for this already

File: ci/kubetest/utils.py [hunk 3]
```diff
@@ -92,18 +119,28 @@ def test_if_posthog_deployments_are_healthy(kube):
...
+
+
+def _exec_subprocess(cmd):
+    cmd_run = subprocess.run(cmd, shell=True, capture_output=True, text=True)
     cmd_return_code = cmd_run.returncode
     if cmd_return_code:
-        pytest.fail("❌ Error while running '{}'. Return code: {}".format(cmd, cmd_return_code))
-    log.debug("✅ Done!")
+        pytest.fail(
+            f"""
```
Comment (guidoiaquinti) [inline ci/kubetest/utils.py:138]: maybe it's GH but doesn't seems to be indented properly

Comment (macobo) [reply ci/kubetest/utils.py:138]: indented by black, so should be OK

File: ci/kubetest/test_clickhouse_custom_settings_profile.py (hunk 2 above)
Comment (macobo) [reply ci/kubetest/test_clickhouse_custom_settings_profile.py:54]: https://kubetest.readthedocs.io/en/latest/reference.html#pod Doesn't seem like it. However I could probably use the HTTP API for this + http_proxy methods. Will experiment!

Comment (macobo) [reply ci/kubetest/test_clickhouse_custom_settings_profile.py:54]: This almost works - ran into a hitch where authentication always fails with this clickhouse-side error:

```
2022.01.25 15:09:23.030568 [ 60 ] {} <Warning> AddressPatterns: Failed to check if the allowed client hosts contain address ::ffff:10.244.0.1. DB::Exception: Cannot getnameinfo(::ffff:10.244.0.1): Name or service not known, code = 198
```

Will need to dig deeper tomorrow.

File: charts/posthog/templates/clickhouse_instance.yaml [hunk 4]
```diff
@@ -12,7 +12,7 @@ spec:
       admin/quota: default
 
     profiles:
-      default/allow_experimental_window_functions: "1"
+      {{- merge dict .Values.clickhouse.profiles .Values.clickhouse.defaultProfiles | toYaml | nindent 6 }}
```
Comment (guidoiaquinti) [inline charts/posthog/templates/clickhouse_instance.yaml:15]: not sure about the ordering (and if something changes) but should we do default first and then overrides?

File: charts/posthog/values.yaml [hunk 5]
```diff
@@ -8,7 +8,7 @@ image:
   sha:
   # -- Posthog image tag, e.g. release-1.31.1
   tag:
-  # -- Default image or tag, e.g. :release-1.31.1
```
Comment (guidoiaquinti) [inline charts/posthog/values.yaml:11]: Why?

Comment (macobo) [reply charts/posthog/values.yaml:11]: Should override one of the two above settings instead. Could make that clear in the comment though

File: ci/kubetest/test_clickhouse_custom_settings_profile.py (hunk 2 above)
Comment (macobo) [reply ci/kubetest/test_clickhouse_custom_settings_profile.py:54]: Dug a bit more - I don't think it's worth investing more here, seems like a deeper clickhouse-operator related bug than worth tackling here.

**Review by guidoiaquinti — APPROVED** [review]
